datafusion/src/physical_plan/window_frames.rs

```diff
@@ -0,0 +1,200 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Window frame
+//!
+//! The frame-spec determines which output rows are read by an aggregate window function. The frame-spec consists of four parts:
+//! - A frame type - either ROWS, RANGE or GROUPS,
+//! - A starting frame boundary,
+//! - An ending frame boundary,
+//! - An EXCLUDE clause.
+
+use crate::error::{DataFusionError, Result};
+use sqlparser::ast::{WindowFrame, WindowFrameBound, WindowFrameUnits};
+
+const DEFAULT_WINDOW_FRAME: WindowFrame = WindowFrame {
+    units: WindowFrameUnits::Range,
+    start_bound: WindowFrameBound::Preceding(None),
+    end_bound: Some(WindowFrameBound::CurrentRow),
+};
+
+fn get_bound_rank(bound: &WindowFrameBound) -> (u8, u64) {
```

Comment:
I don't understand the hard coded numbers in this function. Perhaps giving them names or adding some doc comments would help